packages/datadog-instrumentations/src/helpers/router-helper.js

```diff
@@ -0,0 +1,176 @@
+'use strict'
+
+const shimmer = require('../../../datadog-shimmer')
+
+const routerMountPaths = new WeakMap() // to track mount paths for router instances
+const layerMatchers = new WeakMap() // to store layer matchers
+const appMountedRouters = new WeakSet() // to track routers mounted via app.use()
+
+const METHODS = [...require('http').METHODS.map(v => v.toLowerCase()), 'all']
+
+function joinPath (base, path) {
+  if (!base || base === '/') return path || '/'
+  if (!path || path === '/') return base
+  if (base.endsWith('/') && path.startsWith('/')) return base + path.slice(1)
+  return base + path
+}
+
+// Normalize route definitions coming from Express into a string representation
+function normalizeRoutePath (path) {
+  if (path == null) return null
+  if (path instanceof RegExp) return path.toString()
+  if (typeof path === 'string') return path
```

Comment:
Nit: swap in order of most likely to happen in the real world (I think a string is more likely than an regex, feel free to ignore if I'm wrong):

```suggestion
  if (typeof path === 'string') return path
  if (path instanceof RegExp) return path.toString()
```